packages/teams-js/src/public/marketplace.ts

```diff
@@ -0,0 +1,211 @@
+  /** The cart is created by admin of an organization. */
+  AdminUser = 'AdminUser',
+  /** The cart is created by end user of an organization. */
+  EndUser = 'EndUser',
+}
+
+/**
+ * Represents the status of the cart.
+ *
+ * @beta
+ */
+export enum CartStatus {
+  /** Cart is created but not checked out yet. */
+  Open = 'Open',
+  /** Cart is checked out but not completed yet. */
+  Processing = 'Processing',
+  /**
+   * Indicate checking out is completed and the host should
+   * Response a new cart in the next getCart call.
+   */
+  Processed = 'Processed',
+  /**
+   * Indicate checking out is failed and the host should
+   * Response a new cart in the next getCart call.
+   */
+  Error = 'Error',
+}
+/**
+ * Represents the parameters to update the cart status.
+ *
+ * @beta
+ */
+export interface UpdateCartStatusParams {
+  /** Status of the cart. */
+  cartStatus: CartStatus;
+  /** Extra info to the status. */
+  statusInfo?: string;
+}
+
+/**
+ * Namespace for a vendor app to support a checkout flow by interacting with the marketplace cart in the host.
+ */
+export namespace marketplace {
+  /**
+   * Get the cart object owned by the host to checkout.
+   *
+   * @beta
+   */
+  export function getCart(): Promise<Cart> {
+    return new Promise<Cart>((resolve) => {
+      ensureInitialized(runtime, FrameContexts.content);
+      if (!isSupported()) {
+        throw errorNotSupportedOnPlatform;
+      }
+      resolve(sendAndHandleSdkError('marketplace.getCart'));
+    });
+  }
+  /**
+   * Add or update cart items in the cart owned by the host.
+   *
+   * @param cartItems - A list of cart items, if item id exists, overwrite the item, otherwise add new items to cart.
+   *
+   * @beta
+   */
+  export function addOrUpdateCartItems(cartItems: CartItem[]): Promise<void> {
+    return new Promise<void>((resolve) => {
+      ensureInitialized(runtime, FrameContexts.content);
+      if (!isSupported()) {
+        throw errorNotSupportedOnPlatform;
+      }
+      resolve(sendAndHandleSdkError('marketplace.addOrUpdateCartItems', cartItems));
+    });
+  }
+  /**
+   * Remove cart items from the cart owned by the host.
+   *
+   * @param cartItemIds - A list of cart id, delete the cart item accordingly.
+   *
+   * @beta
+   */
+  export function removeCartItems(cartItemIds: string[]): Promise<void> {
+    return new Promise<void>((resolve) => {
+      ensureInitialized(runtime, FrameContexts.content);
+      if (!isSupported()) {
+        throw errorNotSupportedOnPlatform;
+      }
+      resolve(sendAndHandleSdkError('marketplace.removeCartItems', cartItemIds));
+    });
+  }
+  /**
+   * Update cart status in the cart owned by the host.
+   *
+   * @param updateCartStatusParams
+   *
+   * updateCartStatusParams.cartStatus - cart status.
+   *
+   * updateCartStatusParams.message - extra info to the status.
+   *
+   * @beta
+   */
+  export function updateCartStatus(updateCartStatusParams: UpdateCartStatusParams): Promise<void> {
```

Comment:
consider adding cart id to this. If this is ever service backed you will need to know things like cart id, and it will save you plumbing in the future.